src/ProtectedProxies/SphereXUpgradeableBeacon.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: UNLICENSED
+// (c) SphereX 2023 Terms&Conditions
+
+pragma solidity ^0.8.0;
+
+import {SphereXProtectedProxy} from "../SphereXProtectedProxy.sol";
+import {UpgradeableBeacon} from "@openzeppelin/contracts/proxy/beacon/UpgradeableBeacon.sol";
+
+contract SphereXUpgradeableBeacon is SphereXProtectedProxy, UpgradeableBeacon {
```

Comment:
when I re think of it, why is it a proxy? It should have the interface of a proxy at all...

Reply:
it needs some of the spherex_base functionallity and the protected sigs logic from the proxy, do you think it is better to create another contract like proxy_storage" or something and have SphereXUpgradeableBeacon inherit from it and from base?